p751toolbox/curve.go

```diff
@@ -8,92 +8,20 @@ type ProjectiveCurveParameters struct {
-	var curveParams ProjectiveCurveParameters
-	var t0, t1 ExtensionFieldElement
-	t0.One()                               // = 1
-	t1.Mul(affine_xP, affine_xQ)           // = x_P * x_Q
-	t0.Sub(&t0, &t1)                       // = 1 - x_P * x_Q
-	t1.Mul(affine_xP, affine_xQmP)         // = x_P * x_{Q-P}
-	t0.Sub(&t0, &t1)                       // = 1 - x_P * x_Q - x_P * x_{Q-P}
-	t1.Mul(affine_xQ, affine_xQmP)         // = x_Q * x_{Q-P}
-	t0.Sub(&t0, &t1)                       // = 1 - x_P * x_Q - x_P * x_{Q-P} - x_Q * x_{Q-P}
-	curveParams.A.Square(&t0)              // = (1 - x_P * x_Q - x_P * x_{Q-P} - x_Q * x_{Q-P})^2
-	t1.Mul(&t1, affine_xP)                 // = x_P * x_Q * x_{Q-P}
-	t1.Add(&t1, &t1)                       // = 2 * x_P * x_Q * x_{Q-P}
-	curveParams.C.Add(&t1, &t1)            // = 4 * x_P * x_Q * x_{Q-P}
-	t0.Add(affine_xP, affine_xQ)           // = x_P + x_Q
-	t0.Add(&t0, affine_xQmP)               // = x_P + x_Q + x_{Q-P}
-	t1.Mul(&curveParams.C, &t0)            // = 4 * x_P * x_Q * x_{Q-P} * (x_P + x_Q + x_{Q-P})
-	curveParams.A.Sub(&curveParams.A, &t1) // = (1 - x_P * x_Q - x_P * x_{Q-P} - x_Q * x_{Q-P})^2 - 4 * x_P * x_Q * x_{Q-P} * (x_P + x_Q + x_{Q-P})
-
-	return curveParams
-}
-
-// Compute the j-invariant (not the J-invariant) of the given curve.
-func (curveParams *ProjectiveCurveParameters) JInvariant() ExtensionFieldElement {
-	var v0, v1, v2, v3 ExtensionFieldElement
-	A := &curveParams.A
-	C := &curveParams.C
-	v0.Square(C)           // C^2
-	v1.Square(A)           // A^2
-	v2.Add(&v0, &v0)       // 2C^2
-	v3.Add(&v2, &v0)       // 3C^2
-	v2.Add(&v2, &v2)       // 4C^2
-	v2.Sub(&v1, &v2)       // A^2 - 4C^2
-	v1.Sub(&v1, &v3)       // A^2 - 3C^2
-	v3.Square(&v1)         // (A^2 - 3C^2)^2
-	v3.Mul(&v3, &v1)       // (A^2 - 3C^2)^3
-	v0.Square(&v0)         // C^4
-	v3.Mul(&v3, &const256) // 256(A^2 - 3C^2)^3
-	v2.Mul(&v2, &v0)       // C^4(A^2 - 4C^2)
-	v2.Inv(&v2)            // 1/C^4(A^2 - 4C^2)
-	v0.Mul(&v3, &v2)       // 256(A^2 - 3C^2)^3 / C^4(A^2 - 4C^2)
-	return v0
-}
-
-// Compute cached parameters A + 2C, 4C.
-func (curve *ProjectiveCurveParameters) cachedParams() CachedCurveParameters {
-	var cached CachedCurveParameters
-	cached.Aplus2C.Add(&curve.C, &curve.C)          // = 2*C
-	cached.C4.Add(&cached.Aplus2C, &cached.Aplus2C) // = 4*C
-	cached.Aplus2C.Add(&cached.Aplus2C, &curve.A)   // = 2*C + A
-	return cached
-}
-
-// Compute cached parameters A - 2C, 2C.
-func (curve *ProjectiveCurveParameters) cachedTripleParams() CachedTripleCurveParameters {
-	var cached CachedTripleCurveParameters
-	cached.C2.Add(&curve.C, &curve.C)         // = 2*C
-	cached.Aminus2C.Sub(&curve.A, &cached.C2) // = A- 2*C
-	return cached
+// Stores curve projective parameters equivalent to A/C. Meaning of the
+// values depends on the context. When working with isogenies over
+// subgroup that are powers of:
+// * three then  A=(A+2C)/4; C=(A-2C)/4
+// * four then   A=(A+2C)/4; C=4C
```

Comment:
Where is the denominator of 4 coming from? The Sike spec just defines them as (A+2C : A-2C) and (A+2C : 4C)